diff --git a/libavcodec/rkmppdec.c b/libavcodec/rkmppdec.c
index 2fd42b3e0c..8e86edec41 100644
--- a/libavcodec/rkmppdec.c
+++ b/libavcodec/rkmppdec.c
@@ -161,6 +161,7 @@ static av_cold int rkmpp_init_decoder(AVCodecContext *avctx)
         goto fail;
     }

+    av_log(avctx, AV_LOG_INFO, "RKMPP: checking codec support\n");
     ret = mpp_check_support_format(MPP_CTX_DEC, codectype);
     if (ret != MPP_OK) {
         av_log(avctx, AV_LOG_ERROR, "Codec type (%d) unsupported by MPP\n", avctx->codec_id);
@@ -169,6 +170,7 @@ static av_cold int rkmpp_init_decoder(AVCodecContext *avctx)
     }

     // Create the MPP context
+    av_log(avctx, AV_LOG_INFO, "RKMPP: creating MPP context\n");
     ret = mpp_create(&decoder->ctx, &decoder->mpi);
     if (ret != MPP_OK) {
         av_log(avctx, AV_LOG_ERROR, "Failed to create MPP context (code = %d).\n", ret);
@@ -177,6 +179,7 @@ static av_cold int rkmpp_init_decoder(AVCodecContext *avctx)
     }

     // initialize mpp
+    av_log(avctx, AV_LOG_INFO, "RKMPP: initializing MPP context\n");
     ret = mpp_init(decoder->ctx, MPP_CTX_DEC, codectype);
     if (ret != MPP_OK) {
         av_log(avctx, AV_LOG_ERROR, "Failed to initialize MPP context (code = %d).\n", ret);
@@ -202,7 +205,13 @@ static av_cold int rkmpp_init_decoder(AVCodecContext *avctx)
         goto fail;
     }

-    ret = mpp_buffer_group_get_internal(&decoder->frame_group, MPP_BUFFER_TYPE_ION);
+    av_log(avctx, AV_LOG_INFO, "RKMPP: requesting DRM buffer group\n");
+    ret = mpp_buffer_group_get_internal(&decoder->frame_group, MPP_BUFFER_TYPE_DRM);
+    if (ret) {
+       av_log(avctx, AV_LOG_WARNING, "Failed to retrieve DRM buffer group (code = %d), falling back to ION\n", ret);
+       ret = mpp_buffer_group_get_internal(&decoder->frame_group, MPP_BUFFER_TYPE_ION);
+    }
+
     if (ret) {
        av_log(avctx, AV_LOG_ERROR, "Failed to retrieve buffer group (code = %d)\n", ret);
        ret = AVERROR_UNKNOWN;
